code-challenges.test.js

```diff
@@ -25,35 +28,127 @@ const numbersArray2 = [24, 27, 30, 33, 36]
+// declare a function called allCaps that takes in an array of strings
+// use the .map higher order function to touch every item in the array and then create a new array
+// utilize .charAt(), .toUpperCase(), and .slice() to capitalize only the first letter of the string while still returning the entire string
+// input: an array of strings
+// output: a new array of strings with the first letter of each string capitalized 
+
 const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
-// b) Create the function that makes the test pass.
+const allCaps = (array) => {
+    let newArray= array.map(value => {
+        return value.charAt(0).toUpperCase() + value.slice(1)
+    })
+    return newArray
+}
+console.log(allCaps(randomNouns1))
+// Actual output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+console.log(allCaps(randomNouns2))
+// Actual output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+// RED
+describe("firstVowel", () => {
+    it("logs the index of the first vowel in a given string", () => {
+        expect(firstVowel(vowelTester1)).toEqual(1)
+        expect(firstVowel(vowelTester2)).toEqual(0)
+        expect(firstVowel(vowelTester3)).toEqual(2)
+    })
+})
+
+
+// b) Create the function that makes the test pass.
+
+// GREEN
+// Pseudocode:
+// declare a function called firstVowel that takes in a string
+// use .indexOf() to determine the index of the first value within the string
+// input: a string
+// output: the index of the first vowel within the string
+
 const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
 
-// b) Create the function that makes the test pass.
+const firstVowel = (string) => {
+    let vowels = string.search(/[a, e, i, o, u]/g)
+    return vowels
+}
+console.log(firstVowel(vowelTester1))
+// Actual output: 1
+console.log(firstVowel(vowelTester2))
+// Actual output: 0
+console.log(firstVowel(vowelTester3))
+// Actual output: 2
+
+// Note: I utilized Google as well as the help of one of my classmates to find the solution to this problem after running unto the same blocker continuously. I was struggling to figure out how to log the first instance of any value, rather than the code searching one by one for each value I fed into the array I was calling on. With those resources, I was introduced to the .search(/[]/) method, which will search for the first instance of any given array value rather than searching in the order the values are listed.
```

Comment:
Remember your input data type is a string, so a function that iterates (HOF or for loops), and looks for an index containing vowels (charAt(i), .match, .includes)